### PR TITLE
QPID-8666: [Broker-J] Broker plugin jdbc-provider-bone replacement

### PR DESCRIPTION
**Pull request by dakirily**
This PR is a follow-up to the [PR-235](https://github.com/apache/qpid-broker-j/pull/235) and  [PR-238](https://github.com/apache/qpid-broker-j/pull/238) and addresses JIRA [QPID-8666](https://issues.apache.org/jira/browse/QPID-8666), adding upgrading the broker model version from 9.0 to 9.1

### CODE REVIEW

File: broker-core/src/test/java/org/apache/qpid/server/store/BrokerStoreUpgraderAndRecovererTest.java [hunk 1]
```diff
@@ -898,10 +897,75 @@ public void testContextVariableUpgradeFromBoneCPToHikariCPProvider(final String
...
                 "type", "JDBC",
                 "connectionPoolType", "BONECP",
                 "context", context);
+
+        _brokerRecord.getAttributes().put("context", context);
+
+        final ConfiguredObjectRecord systemConfigRecord = mock(ConfiguredObjectRecord.class);;
+        when(systemConfigRecord.getId()).thenReturn(randomUUID());
+        when(systemConfigRecord.getType()).thenReturn("SystemConfig");
+        when(systemConfigRecord.getAttributes()).thenReturn(Map.copyOf(attributes));
```
[the remark this answers is not in the document]
Comment (dakirily) [reply broker-core/src/test/java/org/apache/qpid/server/store/BrokerStoreUpgraderAndRecovererTest.java]: The test was fixed to use a copy of a context map for each object.

File: broker-core/src/test/java/org/apache/qpid/server/store/VirtualHostStoreUpgraderAndRecovererTest.java [hunk 2]
```diff
@@ -323,29 +323,36 @@ public void testContextVariableUpgradeFromBoneCPToHikariCPProvider(final String
...
         final ConfiguredObjectRecord virtualHostRecord = mock(ConfiguredObjectRecord.class);;
         when(virtualHostRecord.getId()).thenReturn(randomUUID());
         when(virtualHostRecord.getType()).thenReturn("VirtualHost");
         when(virtualHostRecord.getAttributes()).thenReturn(attributes);
 
-        final List<ConfiguredObjectRecord> records = List.of(rootRecord, virtualHostRecord);
+        final ConfiguredObjectRecord virtualHostLoggerRecord = mock(ConfiguredObjectRecord.class);;
+        when(virtualHostLoggerRecord.getId()).thenReturn(randomUUID());
+        when(virtualHostLoggerRecord.getType()).thenReturn("VirtualHostLogger");
+        when(virtualHostLoggerRecord.getAttributes()).thenReturn(attributes);
```
[the remark this answers is not in the document]
Comment (dakirily) [reply broker-core/src/test/java/org/apache/qpid/server/store/VirtualHostStoreUpgraderAndRecovererTest.java]: The test was fixed in the same way as in BrokerStoreUpgraderAndRecovererTest to keep to uniformity.

**Review by gemmellr — COMMENTED** [review]
The change from prior feedback looks good.

I noticed something I didnt earlier when considering it in the context of the new changes.

I was then also looking at the earlier changes again and wonder about the apparent different naming and values between the java code and the web ui code. Its not clear to me they are consistent. Are they meant to differ in the ways they do? E.g:

https://github.com/apache/qpid-broker-j/blob/dae78d991f42c2ca2b85f9e365762bf98bf5e10f/broker-plugins/jdbc-provider-hikari/src/main/java/org/apache/qpid/server/store/jdbc/hikaricp/HikariCPConnectionProvider.java#L51-L52
https://github.com/apache/qpid-broker-j/blob/dae78d991f42c2ca2b85f9e365762bf98bf5e10f/broker-plugins/jdbc-provider-hikari/src/main/java/org/apache/qpid/server/store/jdbc/hikaricp/HikariCPConnectionProviderFactory.java#L41-L44

https://github.com/apache/qpid-broker-j/blob/dae78d991f42c2ca2b85f9e365762bf98bf5e10f/broker-plugins/jdbc-provider-hikari/src/main/java/resources/js/qpid/management/store/pool/hikaricp/add.js#L40-L41
https://github.com/apache/qpid-broker-j/blob/dae78d991f42c2ca2b85f9e365762bf98bf5e10f/broker-plugins/jdbc-provider-hikari/src/main/java/resources/js/qpid/management/store/pool/hikaricp/show.js#L24
https://github.com/apache/qpid-broker-j/blob/dae78d991f42c2ca2b85f9e365762bf98bf5e10f/broker-plugins/jdbc-provider-hikari/src/main/java/resources/js/qpid/management/store/pool/hikaricp/show.js#L71

File: broker-core/src/main/java/org/apache/qpid/server/store/UpgraderHelper.java [hunk 3]
```diff
@@ -79,32 +80,45 @@ public static ConfiguredObjectRecord upgradeConnectionPool(final ConfiguredObjec
...
-                final int maximumPoolSize = newContext.get(MAX_POOL_SIZE_PARAM) != null && partitionCount != 0
-                        ? Integer.parseInt(newContext.get(MAX_POOL_SIZE_PARAM)) * partitionCount : 40;
-                final int minIdle = newContext.get(MIN_IDLE_PARAM) != null && partitionCount != 0
-                        ? Integer.parseInt(newContext.get(MIN_IDLE_PARAM)) * partitionCount : 20;
                 newContext.put(MAX_POOL_SIZE_PARAM, String.valueOf(maximumPoolSize));
                 newContext.put(MIN_IDLE_PARAM, String.valueOf(minIdle));
             }
-            final Map<String, Object> updatedAttributes = new HashMap<>(record.getAttributes());
-            if (BONECP.equals(attributes.get(CP_TYPE)))
+            else if ("Broker".equals(record.getType()))
```
[the remark this answers is not in the document]
Comment (dakirily) [reply broker-core/src/main/java/org/apache/qpid/server/store/UpgraderHelper.java]: Variable name were renamed appropriately.

The upgrader code was changed to make logic for BONECP old type and "Broker" entity the same.

[the remark this answers is not in the document]
Comment (gemmellr) [reply broker-core/src/main/java/org/apache/qpid/server/store/UpgraderHelper.java]: looks good, merged

**Pull request merged**